### ISSUE TITLE
[Bug] Checkout blocks thow error when thousand and decimal separators are identical

### ISSUE THREAD

**Issue by ralucaStan** [woothemes/woocommerce#46045]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When a store uses the same symbol for the `Thousand separator` and `Decimal separator` and a user adds a product to cart with a price bigger then 1000 the checkout blocks don't render the product. In the back the blocks throw an error.

A separate ticket was opened for handling this on the Admin
https://github.com/woocommerce/woocommerce/issues/46047

### Expected behavior

The product should show up.
To fix this bug we need to be defensive in the checkout blocks and remove (not render) the `Thousand separator` in the case the separator characters are similar. 

### Actual behavior

When a merchant repeats the process he sees this error in any of the blocks:

```Unexpected error in: woocommerce/cart-line-items-block

Error: Decimal separator can't be same as thousand separator. thousandSeparator: , (thousandSeparator = {true} is same as thousandSeparator = ",") decimalSeparator: , (default value for decimalSeparator is .)
```
This error [originates](https://github.com/s-yadav/react-number-format/blob/5a4030ca1276ac30f2ad6211ec752a562b278511/src/numeric_format.tsx#L307-L313) from the https://github.com/s-yadav/react-number-format library which we use for formatting the prices. 
p1711634732317289-slack-C8X6Q7XQU

Another approach would be to reconsider the use of https://github.com/s-yadav/react-number-format , considering the prices are showing as expected in the product blocks.

### Steps to reproduce

1. A site uses the same values for the Thousand separator & Decimal separator 
2. A shopper adds a product to the cart with a price bigger then 1000
3. The shopper goes to any of the Cart, Checkout and Mini-Cart blocks; they are empty and the product does not show up
<img width="1333" alt="Screenshot 2024-03-28 at 15 55 18" src="https://github.com/woocommerce/woocommerce/assets/1628454/c61ace56-d8ec-412d-b44d-9d85a2eb3075">

### WordPress Environment

-

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by cat-og** [comment]
Ran into this issue today in 7980072-zd-a8c. This was triggered by this related issue: https://github.com/Automattic/woocommerce-payments/issues/8559 which set their decimal separator to be a comma, despite appearing to be a period in their settings. While  heir thousands separator was set to use a comma in their settings, their cart would not load. 

To get around this on their site, they have updated the thousands separator to use another symbol, but they would like to be updated when this is resolved so they can set that back to use a comma again.

**Comment by m155y** [comment]
Another here: 7987217-zd-a8c

**Comment by ralucaStan** [comment]
> To get around this on their site, they have updated the thousands separator to use another symbol, but they would like to be updated when this is resolved so they can set that back to use a comma again.

@cat-og having the same character is not an official format for any country. I am curious why the merchant needs such a setup. Is this something you can help with?

**Comment by ralucaStan** [comment]
> Another here: 7987217-zd-a8c

@m155y From what I understood they hope to get `1,000.00`, but they introduced `,` for the decimal separator, instead of `.`.


To fix this issue they should have the following setup:
<img width="370" alt="Screenshot 2024-04-04 at 11 32 52" src="https://github.com/woocommerce/woocommerce/assets/1628454/c6df6c89-0fd7-457d-aae2-feac060d9e48">

Note: the official Swedish format is `1 000,00`

**Comment by cat-og** [comment]
> @cat-og having the same character is not an official format for any country. I am curious why the merchant needs such a setup. Is this something you can help with?

@ralucaStan The user does not want to use the same character for both, they'd like to use the format `1,000.00`, but the bug here https://github.com/Automattic/woocommerce-payments/issues/8559 is updating the decimal separator to `,` despite their settings, as though they'd tried to use the format `1,000,00`. This triggered the current bug.

**Comment by thuautp** [comment]
Also reported in 7997200-zen

**Comment by namiokuzono** [comment]
7976972-zd-a8c

**Comment by Robertght** [comment]
8008272-zd-a8c adding this for another HE.

**Comment by shrutimoorthy** [comment]
8050429-zen-a8c

**Comment by ralucaStan** [comment]
A fix is coming with WooCommerce 8.9
